src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java

```diff
@@ -83,8 +84,13 @@
       Module module = buildModule(project);
 
       findResolvedArtifacts(project, allConfigurations).forEach(resolvedArtifact -> {
-        Artifact artifact = new Artifact().setId(resolvedArtifact.getId().getComponentIdentifier().getDisplayName())
-            .setPathname(resolvedArtifact.getFile()).setMonitored(true);
+        ModuleVersionIdentifier artifactId = resolvedArtifact.getModuleVersion().getId();
+
+        Artifact artifact = new Artifact()
+            .setId(artifactId.getGroup() + ":" + artifactId.getName() + ":" + artifactId.getVersion())
+            .setPathname(resolvedArtifact.getFile())
+            .setMonitored(true);
+
```

Comment:
Should there be a unit test to test this part? that the artifact has the expected outcome?

Reply:
> Should there be a unit test to test this part? that the artifact has the expected outcome?

There are already tests checking the output is group:name (artifactId):version

The case the customer presents couldn't be reproduced (getting extra info in version for SNAPSHOT components) using the current approach, but concatenating those fields manually proved to provide a proper result as per the output they provided in the latest reply by mail.